### PR TITLE
ref(page-filters): Use DropdownAutocomplete for TimeRangeSelector

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before:**
Screen Shot 2022-03-05 at 10 47 17 AM
Screen Shot 2022-03-05 at 10 49 42 AM

**After:**
Screen Shot 2022-03-05 at 10 47 52 AM
Screen Shot 2022-03-05 at 10 50 23 AM

Requires https://github.com/getsentry/getsentry/pull/7119

### CODE REVIEW

File: static/app/components/organizations/timeRangeSelector/index.tsx [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-import DropdownMenu, {GetActorPropsFn} from 'sentry/components/dropdownMenu';
+import DropdownAutoComplete from 'sentry/components/dropdownAutoComplete';
+import {Item} from 'sentry/components/dropdownAutoComplete/types';
+import {GetActorPropsFn} from 'sentry/components/dropdownMenu';
 import HookOrDefault from 'sentry/components/hookOrDefault';
 import HeaderItem from 'sentry/components/organizations/headerItem';
 import MultipleSelectorSubmitRow from 'sentry/components/organizations/multipleSelectorSubmitRow';
+import PageFilterPinButton from 'sentry/components/organizations/pageFilters/pageFilterPinButton';
 import DateRange from 'sentry/components/organizations/timeRangeSelector/dateRange';
-import SelectorItems from 'sentry/components/organizations/timeRangeSelector/dateRange/selectorItems';
```
Comment (davidenwang) [inline static/app/components/organizations/timeRangeSelector/index.tsx:10]: do we need to delete this old selectorItems?

File: static/app/components/pageTimeRangeSelector.tsx [hunk 2]
```diff
@@ -1,66 +1,30 @@
...
         label={<DropdownLabel>{t('Date Range:')}</DropdownLabel>}
-        onToggleSelector={isOpen => setIsCalendarOpen(isOpen)}
         relativeOptions={DEFAULT_RELATIVE_PERIODS}
         {...props}
       />
     </DropdownDate>
   );
 }
 
-const DropdownDate = styled(Panel)<{isCalendarOpen: boolean}>`
```
Comment (davidenwang) [inline static/app/components/pageTimeRangeSelector.tsx:30]: we still might need to preserve some of these styles for the standalone date filter on release details and team stats
image

**Review by davidenwang — COMMENTED** [review]
Dang looking really good, just some small things and also noticed this bug where closing the time selector by clicking on the dropdown button after selecting an absolute date does not work

Kapture 2022-03-08 at 15 46 09

File: static/app/components/organizations/timeRangeSelector/index.tsx (hunk 1 above)
Comment (vuluongj20) [reply static/app/components/organizations/timeRangeSelector/index.tsx:10]: done! https://github.com/getsentry/sentry/pull/32349/commits/ae280483d2cbdebd581a029f6cd5cd1a88b0b0cb

File: static/app/components/pageTimeRangeSelector.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/components/pageTimeRangeSelector.tsx:30]: fixed in https://github.com/getsentry/sentry/pull/32349/commits/e0a40ac3c351d558a37f0b11a81528f52ce86664

**Comment by vuluongj20** [comment]
> Dang looking really good, just some small things and also noticed this bug where closing the time selector by clicking on the dropdown button after selecting an absolute date does not work
> 
> Kapture 2022-03-08 at 15 46 09

after some digging around, looks like we just need to set `allowActorToggle` to true the https://github.com/getsentry/sentry/pull/32349/commits/96f0346b59754e8213f582c9aa80b41369cb6390

**Review by davidenwang — APPROVED** [review]

**Pull request merged**